### ISSUE TITLE
Remove isMountComplete check, and only start ProjFS once VFS4G is ready to service requests

### ISSUE THREAD

**Issue by wilbaker** [Microsoft/VFSForGit#793]
Porting from old backlog (see User Story 1258754).

There should be no need to start the virtualization instance before VFS4G is ready.

**Comment by wilbaker** [comment]
@jeschu1 @jrbriggs if we pick this work item up it eliminates the need for ProjFS on Mac to allow providers to return an error from the PreConvertToFull callback (see #791 for additional context)

**Comment by jeschu1** [comment]
I won't open a separate issue for error handling on ModifiedFiles/PreConvertToFull
